### PR TITLE
[CFE-475] Add code snippets for components

### PR DESCRIPTION
**Pull request by danielbrzn**
This PR adds code snippets/classes (where applicable) for elements and components of the style guide.

Other notable changes:
- Refactored icon rendering (render the template using an array instead of repeating HTML with different classes)

- Add syntax highlighting, together with a copy code button for easier development

- Removed text preview function as per new Style Guide design

- Merged SVG and Sprite Icons into a unified section

- Split Typography into Fonts and Text Styles as per new Style Guide design

- Display SVG Icons within their respective categories

Todo:
- ~Add code snippets for video section~
- ~Remove trailing whitespace~

### CODE REVIEW

File: app/views/sephora_style_guide/home/_accordion.slim [hunk 1]
```diff
@@ -113,3 +149,74 @@ article
...
+            .card.card-light
+              #skin-concerns-heading.card-header(data-toggle='collapse' data-target='#skin-concerns-collapse' role='tab')
+                i.collapse-arrow
+                h4.h4 SKINCARE
+
+              #skin-concerns-collapse.collapse.show(aria-labelledby='skin-concerns-heading' role='tabpanel' data-parent='#sidebar-accordion')
+                .card-block
+                  ul
+                    li
+                      a(href='#0') Cleanser & Exfoliator
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_accordion.slim]: Shouldn't this be `&amp;`

File: app/views/sephora_style_guide/home/_accordion.slim [hunk 2]
```diff
@@ -34,6 +34,42 @@ article
...
             | The Dursleys had everything they wanted, but they also had a secret, and their greatest fear was that somebody would discover it. They didn’t think they could bear it if anyone found out about the Potters. Mrs. Potter was Mrs. Dursley’s sister, but they hadn’t met for several years; in fact, Mrs. Dursley pretended she didn’t have a sister, because her sister and her good-for-nothing husband were as unDursleyish as it was possible to be. The Dursleys shuddered to think what the neighbors would say if the Potters arrived in the street. The Dursleys knew that the Potters had a small son, too, but they had never even seen him. This boy was another good reason for keeping the Potters away; they didn’t want Dudley mixing with a child like that.
 
+    br
+    h3.h3 Code Snippet
+
+    h4.h4 Pug
+    pre
+      code.language-pug
+        |
+          #default-accordion.accordion(aria-multiselectable='true' role='tablist')
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_accordion.slim]: ids should be after classes

File: app/views/sephora_style_guide/home/_accordion.slim [hunk 3]
```diff
@@ -34,6 +34,42 @@ article
...
+
+            .card.card-primary
+              #heading-one.card-header.cursor(role='tab' data-toggle='collapse' data-target='#collapse-one' aria-expanded="true" aria-controls="collapse-one")
+                h3.h3.mb-0
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_accordion.slim]: what is the `mb-0` for? Is it essential to the markdown of this?

File: app/views/sephora_style_guide/home/_accordion.slim [hunk 4]
```diff
@@ -34,6 +34,42 @@ article
...
+    h3.h3 Code Snippet
+
+    h4.h4 Pug
+    pre
+      code.language-pug
+        |
+          #default-accordion.accordion(aria-multiselectable='true' role='tablist')
+
+            .card.card-primary
+              #heading-one.card-header.cursor(role='tab' data-toggle='collapse' data-target='#collapse-one' aria-expanded="true" aria-controls="collapse-one")
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_accordion.slim]: double quotes

File: app/views/sephora_style_guide/home/_accordion.slim [hunk 5]
```diff
@@ -34,6 +34,42 @@ article
...
+                h3.h3.mb-0
+                  | SIGN IN / REGISTER
+
+              #collapse-one.collapse.show(aria-labelledby='heading-one' role='tabpanel' data-parent='#default-accordion')
+                .card-block
+                  | Mr. and Mrs. Dursley, of number four, Privet Drive, were proud to say that they were perfectly normal, thank you very much. They were the last people you’d expect to be involved in anything strange or mysterious, because they just didn’t hold with such nonsense.
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_accordion.slim]: Usually the code snippet should serve as copy-pastable base which doesn't need much changes (apart from additions of course), so placeholder text should be limited, or show exactly what needs to change, for example use keywords like `CHANGEME`).

File: app/views/sephora_style_guide/home/_forms.slim [hunk 6]
```diff
@@ -35,6 +35,43 @@ article#forms
...
       input#email5.form-control type='email' value='testemail' disabled=true
+    
+    br
+    h4.h4 pug
+    pre
+      code.language-pug
+        |
+          .form-group.floating-labels
+            label.control-label(for='email') Email (Empty Field)
+            input#email.form-control(type='email')
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_forms.slim]: the `id` here is redundant as it doesn't serve a purpose in terms of functionality.

File: app/views/sephora_style_guide/home/_forms.slim [hunk 7]
```diff
@@ -77,6 +114,71 @@ article#forms
...
+            label.control-label(for='textarea1') Textarea
+            textarea#textarea1.form-control
+
+          .form-group.floating-labels.textarea.filled.focused
+            label.control-label(for='textarea') Textarea
+            textarea#textarea.form-control
+              = 'Now led tedious shy lasting females off. Dashwood marianne in of entrance be on wondered possible building. Wondered sociable he carriage in speedily margaret. Up devonshire of he thoroughly insensible alteration. An mr settling occasion insisted distance ladyship so. Not attention say frankness intention out dashwoods now curiosity. Stronger ecstatic as no judgment daughter speedily thoughts. Worse downs nor might she court did nay forth these. Now led tedious shy lasting females off. Dashwood marianne in of entrance be on wondered possible building. Wondered sociable he carriage in speedily margaret. Up devonshire of he thoroughly insensible alteration. An mr settling occasion insisted distance ladyship so. Not attention say frankness intention out dashwoods now curiosity. Stronger ecstatic as no judgment daughter speedily thoughts. Worse downs nor might she court did nay forth these.'
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_forms.slim]: Same comment about placeholder text.

File: app/views/sephora_style_guide/home/_pagination.slim [hunk 8]
```diff
@@ -4,7 +4,7 @@ article#pagination
   section
     h3.h3 Top
     br
-    label Less then 3 pages
+    label Less than 3 pages
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_pagination.slim]: LOL 😂 Thanks for spotting though!

File: app/views/sephora_style_guide/home/_pagination.slim [hunk 9]
```diff
@@ -83,3 +83,26 @@ article#pagination
...
+        nav.pagination
+          a.page.prev
+          a.page 1
+          a.page.continue ...
+          a.page.current 7
+          a.page 8
+          a.page 9
+          a.page.continue ...
+          a.page 12
+          a.page.next
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_pagination.slim]: Missing final new line. If you are not aware, EditrorConfig should fix these automatically for you.

File: app/views/sephora_style_guide/home/_sprite_icons.slim [hunk 10]
```diff
@@ -1,169 +1,16 @@
...
...":"icon_50x30"},{"name":"Convenience Stores","class":"cstore","size":"icon_50x30"},{"name":"Mandiri","class":"mandiri","size":"icon_70x30"},{"name":"PayPal","class":"paypal","size":"icon_70x30"},{"name":"VISA","class":"visa","size":"icon_70x30"},{"name":"RHB Now","class":"rhbnow","size":"icon_70x30"},{"name":"MEPF Currency","class":"mepf","size":"icon_70x30"},{"name":"Dragonpay","class":"dragonpay","size":"icon_70x30"},{"name":"Cash On Delivery","class":"cod","size":"icon_70x30"},{"name":"BAYAR DI TEMPAT","class":"bayar","size":"icon_70x30"},{"name":"AfterPay","class":"afterpay","size":"icon_70x30"},{"name":"Hongleong Connect","class":"connect","size":"icon_70x30"},{"name":"PAYATALL","class":"payatall","size":"icon_70x30"},{"name":"maybank2u.com","class":"maybank","size":"icon_100x30"},{"name":"Hong Leong Bank","class":"hongleong","size":"icon_100x30"},{"name":"CIMB Clicks","class":"cimbclick","size":"icon_100x30"},{"name":"Permata Bank","class":"permata","size":"icon_100x30"},{"name":"Verified by VISA","class":"visa","size":"icon_90x40"},{"name":"MasterCard SecureCode","class":"mastercard","size":"icon_90x40"},{"name":"AMERICAN EXPRESS SafeKey","class":"amex","size":"icon_90x40"}]
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_sprite_icons.slim]: Nice addition but would be more readable if it was better formatted. You can just make a ruby block like so:
```slim
ruby:
  sprite_icons = [
    {},
    {},
    # ...
  ]
```

File: app/views/sephora_style_guide/home/_sprite_icons.slim [hunk 11]
```diff
@@ -1,169 +1,16 @@
...
-    .icon_container
-      .sprite-icon.icon_90x40.amex
-      span AMERICAN EXPRESS SafeKey
+    - sprite_icons.each do | icon |
+      .icon_container.col-12.col-md-6.col-lg-4
+        .helper
+        .sprite-icon class="#{ icon[:class] }" class="#{ icon[:size] }"
+        div.icon-data
+          p #{ icon[:name] }
+          p.subtext .sprite-icon.#{ icon[:class] }.#{ icon[:size] }
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_sprite_icons.slim]: No need for spaces within `{}` when used for interpolation.

File: app/views/sephora_style_guide/home/_svg_icons.slim [hunk 12]
```diff
@@ -1,22 +1,90 @@
...
+        div.icon-data
+          p #{ icon[:name] }
+          p.subtext .svg-icon.#{ icon[:class] }
+    h3.h3 My Account
+    - my_account.each do | icon |
+      .icon_container.col-12.col-md-6.col-lg-4
+        .helper
+        .svg-icon class="#{ icon[:class] }"
+        div.icon-data
+          p #{ icon[:name] }
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_svg_icons.slim]: If it's just a single argument that needs to be evaluated as ruby, you can do `p= icon[:name]` (Can't recall if there is a space between the `p` and the `=`)

File: app/views/sephora_style_guide/home/_typography.slim [hunk 13]
```diff
@@ -25,30 +25,40 @@ article#typography
...
+          td
+            p.text-avalon-bold style='font-size: 18px;' .text-avalon-bold
         tr
           td
             p.text-avalon-demi style='font-size: 18px;' Avalon Demi
           td
             p.text-avalon-demi v-bind:style="{fontSize: fontSize + 'px'}"
               ' {{ text }}
+          td
+            p.text-avalon-demi style='font-size: 18px;' .text-avalon-demi
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_typography.slim]: possible to add the style to some class? Also the classes need not be in the exact font that they represent. And would be nice to show the `include` mixins when added to CSS.

File: app/views/sephora_style_guide/home/_pagination.slim (hunk 9 above)
Comment (danielbrzn) [reply app/views/sephora_style_guide/home/_pagination.slim]: Somehow it was only working for the main repo.. have it enabled now for the style guide.

File: app/views/sephora_style_guide/home/_accordion.slim (hunk 2 above)
Comment (danielbrzn) [reply app/views/sephora_style_guide/home/_accordion.slim]: Will change it in the snippet and in the component

File: app/views/sephora_style_guide/home/_accordion.slim (hunk 3 above)
Comment (danielbrzn) [reply app/views/sephora_style_guide/home/_accordion.slim]: I just copied it as it was there. However, it doesn't seem to apply anything. Will remove

File: app/views/sephora_style_guide/home/_forms.slim (hunk 6 above)
Comment (danielbrzn) [reply app/views/sephora_style_guide/home/_forms.slim]: It allows the field to be focused when the label is clicked

File: app/views/sephora_style_guide/home/_sprite_icons.slim (hunk 10 above)
Comment (danielbrzn) [reply app/views/sephora_style_guide/home/_sprite_icons.slim]: Fixed

**Comment by danielbrzn** [comment]
Would prefer to merge this before https://github.com/luxola/sephora-style-guide/pull/71 as the sidebar has to reference the new sections added in this PR.

File: app/views/sephora_style_guide/home/_modals.slim [hunk 14]
```diff
@@ -37,16 +37,32 @@ article#modals
...
+          th Options
+          th Description
+      tbody.text-center
+        tr
+          td .modal-xx
+          td sm, lg
+          td Controls the size of the modal. Add this class to the .modal-dialog element
+
+
+
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_modals.slim]: Extra lines

File: app/views/sephora_style_guide/home/_typography.slim [hunk 15]
```diff
@@ -1,110 +1,96 @@
...
+    .page-title
+      ' {{ text }}
+
+    h3.h3 .page-title-italic
+    .page-title-italic
+      ' {{ text }}
+
+    h3.h3 .h1
+    .h1
+      ' {{ text }}
```
Comment (zamakkat) [inline app/views/sephora_style_guide/home/_typography.slim]: This can be changed to just `.h1 = text`

Comment (zamakkat) [reply app/views/sephora_style_guide/home/_typography.slim]: Oops sorry I am getting languages mixed up 😅 Ignore me then.

**Pull request merged**